### ISSUE TITLE
Can't locate the Solution Directory

### ISSUE THREAD

**Issue by xtergs** [dansiegel/Mobile.BuildTools#37]
Thank you for this great tool. I'm trying to set up build on AppCenter, but it's always failing with error 

```
Build started 7/1/2018 3:59:17 PM.
Project "/Users/vsts/agent/2.134.2/work/1/s/Test_MobileTools/Test_MobileTools.Android/Test_MobileTools.Android.csproj" on node 1 (PackageForAndroid target(s)).
_CreatePropertiesCache:
  Creating directory "obj/Release/".
TemplateManifest:
  Checking for Platform Template Configurations to Copy
  SdkShortFrameworkIdentifier: ''
  Prefix: 'Manifest_'
  Token: ''
  ManifestTemplatePath: '/Users/vsts/agent/2.134.2/work/1/s/Test_MobileTools/build/AndroidTemplateManifest.xml'
  ManifestOutputPath: '/Users/vsts/agent/2.134.2/work/1/s/Test_MobileTools/Test_MobileTools.Android/Properties/AndroidManifest.xml'
/Users/vsts/.nuget/packages/mobile.buildtools/1.3.0.599/build/Mobile.BuildTools.targets(35,5): warning : Could not locate template manifest at '/Users/vsts/agent/2.134.2/work/1/s/Test_MobileTools/build/AndroidTemplateManifest.xml' [/Users/vsts/agent/2.134.2/work/1/s/Test_MobileTools/Test_MobileTools.Android/Test_MobileTools.Android.csproj]
_SetLatestTargetFrameworkVersion:
  Found Java SDK version 1.8.0.
/Library/Frameworks/Mono.framework/External/xbuild/Xamarin/Android/Xamarin.Android.Common.targets(602,2): error : AndroidManifest file does not exist [/Users/vsts/agent/2.134.2/work/1/s/Test_MobileTools/Test_MobileTools.Android/Test_MobileTools.Android.csproj]
Done Building Project "/Users/vsts/agent/2.134.2/work/1/s/Test_MobileTools/Test_MobileTools.Android/Test_MobileTools.Android.csproj" (PackageForAndroid target(s)) -- FAILED.

Build FAILED.
```

The path ManifestTemplatePath looks good but anyway AppCenter can't find this file. I tried to build your project, and it's can generate manifest.
I just created empty forms project, add nuget package and copy build folder and directory.props from your repository. Here's my test [repo](https://github.com/xtergs/Test_MobileTools)

**Comment by xtergs** [comment]
@dansiegel I found the issue. It looks like property $(SolutionDir) resolved incorrectly when csproj built directly. 
In my case I created default forms solution and vs placed all projects in "Test_MobileTools" folder. And msbuild trying to find "build" folder in "Test_MobileTools".  But when I renamed this folder to "src" as it in your repository it started to work.

So repository structure should be next:
```
.git\
build\
src\project.core\*.csproj
src\project.droid\*.csproj
Directory.build.props
*.sln
```

**Comment by dansiegel** [comment]
This is less of a bug with the Mobile.BuildTools and more of a limitation caused by the Android Build Task in VSTS (which App Center relies on under the hood). This forces the use of the csproj over a Solution File, and a lot of times developers end up following the Project first approach on iOS. 

To fix this issue the Mobile.BuildTools will need to ensure that the SolutionDirectory is evaluated and set if it is undefined or empty.

**Comment by stevenengland** [comment]
So for now, one could either use this workaround (put into *Droid.csproj)
`<BuildResourcesDirRootPath>$(ProjectDir)..\..\build</BuildResourcesDirRootPath>`
or use a yaml build with the SolutionDir set as an environment var:
```
- task: XamarinAndroid@1
  displayName: 'Build Xamarin.Android project **/*Droid*.csproj'
  inputs:
    projectFile: '**/*Droid*.csproj'
    outputDirectory: '$(build.binariesdirectory)/$(BuildConfiguration)'
    configuration: '$(BuildConfiguration)'
    msbuildArguments: '/verbosity:detailed'
  env:
    SolutionDir: $(Build.Repository.LocalPath)
```

**Comment by dansiegel** [comment]
I'm going to close this issue. This will be fixed by issue #72

**Comment by dansiegel** [comment]
This should be fixed by the new Mobile.BuildTools initialization task in the v2.0 beta when it comes out.
